### PR TITLE
MSI: install Sigil-vNext 4.8.41 into the GAC

### PR DESCRIPTION
**Pull request by lucaspimentel**
Install Sigil-vNext 4.8.41 into the GAC in addition to Sigil 4.7.

```
C:\Windows\Microsoft.NET\assembly\GAC_MSIL\Sigil>dir

02/21/2020  08:25 AM    <DIR>          v4.0_4.7.0.0__2d06c3494341c8ab
02/21/2020  08:25 AM    <DIR>          v4.0_4.8.41.0__2d06c3494341c8ab
```

Tested with apps running on .NET Framework 4.8 and they now correctly load Sigil from the GAC.

Before:
```
Profile attached: True
Path to Datadog.Trace.dll: C:\Windows\Microsoft.Net\assembly\GAC_MSIL\Datadog.Trace\v4.0_1.13.0.0__def86d061d0d2eeb\Datadog.Trace.dll
Path to Datadog.Trace.ClrProfiler.Managed.dll: C:\Windows\Microsoft.Net\assembly\GAC_MSIL\Datadog.Trace.ClrProfiler.Managed\v4.0_1.13.0.0__def86d061d0d2eeb\Datadog.Trace.ClrProfiler.Managed.dll
Path to Sigil.dll: C:\Program Files\Datadog\.NET Tracer\net461\Sigil.dll
```

After:
```
Profile attached: True
Path to Datadog.Trace.dll: C:\Windows\Microsoft.Net\assembly\GAC_MSIL\Datadog.Trace\v4.0_1.13.0.0__def86d061d0d2eeb\Datadog.Trace.dll
Path to Datadog.Trace.ClrProfiler.Managed.dll: C:\Windows\Microsoft.Net\assembly\GAC_MSIL\Datadog.Trace.ClrProfiler.Managed\v4.0_1.13.0.0__def86d061d0d2eeb\Datadog.Trace.ClrProfiler.Managed.dll
Path to Sigil.dll: C:\Windows\Microsoft.Net\assembly\GAC_MSIL\Sigil\v4.0_4.8.41.0__2d06c3494341c8ab\Sigil.dll
```

### CODE REVIEW

File: deploy/Datadog.Trace.ClrProfiler.WindowsInstaller/Files.Managed.Net461.GAC.wxs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  <Fragment>
+    <ComponentGroup Id="Files.Managed.Net461.GAC" Directory="net461.GAC">
+      <Component Win64="$(var.Win64)">
+        <File Id="net461_GAC_Sigil.dll"
+              Source="$(var.ManagedDllPath)\net461\Sigil.dll"
+              KeyPath="yes" Checksum="yes" Assembly=".net"/>
+      </Component>
+    </ComponentGroup>
+  </Fragment>
+</Wix>
```
Comment (colin-higgins) [inline deploy/Datadog.Trace.ClrProfiler.WindowsInstaller/Files.Managed.Net461.GAC.wxs]: Can this be managed in `SyncMsiContent`?
The `Files.Managed.Net45.GAC.wxs` is managed through there, though it is a little different as this is for a single dll.

Comment (lucaspimentel) [reply deploy/Datadog.Trace.ClrProfiler.WindowsInstaller/Files.Managed.Net461.GAC.wxs]: Would be great if we could automate it, but not sure how that would work. `Sigil` is a special case.

Comment (lucaspimentel) [reply deploy/Datadog.Trace.ClrProfiler.WindowsInstaller/Files.Managed.Net461.GAC.wxs]: On second though, until we have more special cases like `Sigil`, there's isn't really anything to automate.

**Comment by C-Howard** [comment]
@lucaspimentel Could you provide some information on why this change was made please? Neither this ticket or the release notes seem to list the issue being fixed.

**Comment by lucaspimentel** [comment]
TL;DR: we already install many assemblies into the GAC and this one should have always been included. This PR fixes that oversight.

More context: [For a while now](https://github.com/DataDog/dd-trace-dotnet/pull/443) we have included 2 versions of Sigil in the Windows installer:
- [Sigil](https://github.com/kevin-montrose/Sigil) v4.7.0 for apps on .NET Framework 4.5-4.5.1
- [Sigil-vNext](https://github.com/arlm/Sigil-vNext) 4.8.41 for apps on .NET Framework 4.5.2 and above, and apps on .NET Core

Why two versions? The original `Sigil` project had not seen any updates for years (until very recently), so `Sigil-vNext` was forked to fix a few bugs. However, that fork also dropped support .NET Framework 4.5. The .NET Tracer [takes a dependency](https://github.com/DataDog/dd-trace-dotnet/blob/81cc1c2dedb61918920dd160d2658111cdef7613/src/Datadog.Trace.ClrProfiler.Managed/Datadog.Trace.ClrProfiler.Managed.csproj#L50-L56) on the older version of Sigil when targeting `net45`, and uses the newer version on all other targets.

Why use the GAC at all if we can use the `DD_DOTNET_TRACER_HOME` directory as fallback? Having the Tracer assemblies and their dependencies in the GAC is required for them to be loaded domain-neutral in ASP.NET applications. This in turn helps avoid [certain assembly-sharing issues](https://github.com/DataDog/dd-trace-dotnet/issues/592) when ASP.NET apps share the same Application Pool in IIS.

Hope that helps!
